lib/jxl/encode.cc

```diff
@@ -473,6 +486,10 @@ JxlEncoderStatus JxlEncoderSetBasicInfo(JxlEncoder* enc,
   if (!enc->metadata.size.Set(info->xsize, info->ysize)) {
     return JXL_ENC_ERROR;
   }
+  if (CheckValidBitdepth(info->bits_per_sample,
```

Comment:
I don't know if this is correct or flipped, but it's probably better to add an explicit `JXL_ENC_SUCCESS !=`.